### PR TITLE
Complete BigQueryNumeric work

### PR DESCRIPTION
**Pull request by jskeet**
(Review each commit separately, as the first only affects Firestore)

### CODE REVIEW

**Comment by jskeet** [comment]
(I've checked, and we can merge to master when we're ready to. That'll happen separately to this PR though, and we'll want another round of review.)

**Comment by jskeet** [comment]
Not sure why the Travis build is failing - that's odd. Will fix...

**Comment by jskeet** [comment]
Whoops - found it. Should be okay now.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryNumericTest.cs [hunk 1]
```diff
@@ -130,6 +193,177 @@ public void ToDecimal_Overflow(LossOfPrecisionHandling handling, bool negate)
...
+        }
+
+        [Theory]
+        [InlineData("0")]
+        [InlineData("1")]
+        [InlineData("-1")]
+        [InlineData(MaxText)]
+        [InlineData(MinText)]
+        [InlineData(EpsilonText)]
+        public void CompareTo_Equal(string text)
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryNumericTest.cs]: Is it worth checking that "0" == "-0"?

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryNumeric.cs [hunk 2]
```diff
@@ -39,20 +38,43 @@ public struct BigQueryNumeric : IEquatable<BigQueryNumeric>, IComparable<BigQuer
...
+
+        /// <summary>
+        /// The minimum valid value for <see cref="BigQueryNumeric"/>, equal to -99999999999999999999999999999.999999999.
+        /// </summary>
+        public static BigQueryNumeric MinValue { get; } = new BigQueryNumeric(s_minValue);
+
+        /// <summary>
+        /// The smallest positive value for <see cref="BigQueryNumeric"/>, equal to 0.000000001.
+        /// </summary>
+        public static BigQueryNumeric Episilon { get; } = new BigQueryNumeric(1);
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryNumeric.cs]: Spelling of `Epsilon` ;)

**Comment by jskeet** [comment]
Will merge on green, then create a new PR to merge the branch into master.

**Comment by jskeet** [comment]
I suspect Travis failures were due to differences between bigquery-numeric and master. I've rebased both bigquery-numeric and the branch for this PR... that will cause a bit of Travis churn, but we should then be good to go *and* good to create a new PR to merge bigquery-numeric into master.

**Pull request merged**